client/src/main/java/org/apache/crail/conf/CrailConfiguration.java

```diff
@@ -39,6 +39,17 @@ public CrailConfiguration() throws IOException {
 		mergeProperties(properties);
 	}
 
+  // This constructor is useful, when a client does
+  // not have access to a file system, but only needs the configuration
+  // of the namenode address. As an example, an OpenWHisk function
+  // does not have a filesystem associated and therefore cannot
+  // read the namenode address from any configuration file.
+  public CrailConfiguration(String nameNodeAddress) throws IOException {
+    conf = new ConcurrentHashMap<>();
+    conf.put("crail.namenode.address", nameNodeAddress);
```

Comment:
I don't mind adding this line. But this really only addresses the namenode parameter (obviously a key parameter). Would be nice to have a more flexible solution where either we can load the config also from different sources, or allow the user to create a plain config and then set certain properties by hand in the case where CRAIL_HOME is not set and properties cannot be loaded.

Reply:
Thanks for the feedback.

I agree, it is a simplistic change and a more generic solution would definitely be better.
One thing I could do is starting with this simple change for now and at the same time create and assign a ticket to myself and come up with a more generic solution.

Reply:
Why not instead make a constructor which creates an empty configuration? and the client can then set all the configuration parameters to its liking?

Reply:
I would be fine with this idea as well.

The easiest way would be to keep the already existing constructor CrailConfiguration(), but move
the call of loadProperties into a public function, which can optionally be called by the user,
if hte wants to read the configuration file.

If the user choses not to load the configuration file, he could just use the existing set() method.

If this is fine with everyone, I will change it accordingly.

Reply:
Sounds good. Alternatively a static function: createFromConfigurationFile() or similar, but I don't have a strong opinion about either.

Reply:
Anything is fine as long as we are not breaking existing applications and benchmarks that assume the properties to be loaded when calling the empty constructor.

Reply:
Good point, actually I was thinking about that as well... If we change it this way, the default empty  constructor would not load the properties, which will break existing code. This is why I originally introduced a different (simplistic) constructor. With this one we would not have the problem of breaking or needing to change existing code.

Reply:
This works both ways though. We can leave the constructor as is but introduce a new static public method `createEmtpy` or similar.
Anyway I personally am not to concerned about breaking things we have done this numerous times in the past. New version, new API is fine for me.